risc0/circuit/rv32im/src/prove/emu/preflight/mod.rs

```diff
@@ -872,31 +872,43 @@ impl<'a> BigInt2Witness<'a> {
 }
 
 impl<'a> bibc::BigIntIO for BigInt2Witness<'a> {
-    fn load(&mut self, arena: u32, offset: u32, count: u32) -> Result<BigUint> {
+    fn load(&mut self, arena: u32, offset: u32, count: u32) -> Result<UBig> {
         // tracing::debug!("load(arena: {arena}, offset: {offset}, count: {count})");
         let base = ByteAddr(self.preflight.peek_register(arena as usize)?);
         let addr = base + offset * BIGINT2_WIDTH_BYTES as u32;
         let bytes = self
             .preflight
             .peek_region(addr.waddr(), count / WORD_SIZE as u32)?;
-        Ok(BigUint::from_bytes_le(&bytes))
+        Ok(UBig::from_le_bytes(&bytes))
     }
 
-    fn store(&mut self, arena: u32, offset: u32, count: u32, value: &BigUint) -> Result<()> {
+    fn store(&mut self, arena: u32, offset: u32, count: u32, value: &UBig) -> Result<()> {
         let base = ByteAddr(self.preflight.peek_register(arena as usize)?);
         let addr = base + offset * BIGINT2_WIDTH_BYTES as u32;
         // tracing::debug!(
         //     "store(arena: {arena}, offset: {offset}, count: {count}, value: {value}, base: {base:?}, addr: {addr:?})"
         // );
         let addr = addr.waddr();
-        let mut words = value.to_u32_digits();
-        let word_count = count as usize / WORD_SIZE;
-        if words.len() < word_count {
-            words.resize(word_count, 0);
+        let mut value_bytes = value.to_le_bytes();
+        // Word-align our byte array in case of a small value
+        while 0 != value_bytes.len() % 4 {
+            value_bytes.push(0);
         }
-        ensure!(words.len() == word_count);
-        for (i, word) in words.iter().enumerate() {
-            self.witness.insert(addr + i, *word);
+        let word_count = count as usize / WORD_SIZE;
+        for i in 0..word_count {
+            let byte_offset = i * 4;
+            let mut word = 0_u32;
+            if byte_offset < value_bytes.len() {
+                assert!(byte_offset + 3 < value_bytes.len());
+                let word_bytes: [u8; 4] = [
+                    value_bytes[byte_offset],
+                    value_bytes[byte_offset + 1],
+                    value_bytes[byte_offset + 2],
+                    value_bytes[byte_offset + 3],
+                ];
```

Comment:
```suggestion
        let word_count = count as usize / WORD_SIZE;
        for i in 0..word_count {
            let byte_offset = i * WORD_SIZE;
            let mut word = 0_u32;
            if byte_offset < value_bytes.len() {
                assert!(byte_offset + 3 < value_bytes.len());
                let word_bytes: [u8; WORD_SIZE] = [
                    value_bytes[byte_offset],
                    value_bytes[byte_offset + 1],
                    value_bytes[byte_offset + 2],
                    value_bytes[byte_offset + 3],
                ];
```

Reply:
great idea - I'll update the assert, too